### PR TITLE
BIGTOP-4063: Adapt bigtop-select to Python 3

### PR DESCRIPTION
**Pull request by timyuer**
### Description of PR
https://issues.apache.org/jira/browse/BIGTOP-4063

### For code changes:

- [ ] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'BIGTOP-3638. Your PR title ...')?
- [ ] Make sure that newly added files do not have any licensing issues. When in doubt refer to https://www.apache.org/licenses/

### CODE REVIEW

**Comment by timyuer** [comment]
Tested successful on rocky8-x86_64, redhat7-x86_64, openEuler-22.04-x86_64, ubuntu-22.04-x86_64.

**Comment by timyuer** [comment]
Refer:
https://github.com/apache/ambari/pull/3751

File: bigtop-packages/src/common/bigtop-select/distro-select [hunk 1]
```diff
@@ -154,7 +154,7 @@ def listPackages( packages ):
   if packages == None:
     packages = leaves
 
-  packages.sort()
+  sorted(packages)
```
Comment (sekikn) [inline bigtop-packages/src/common/bigtop-select/distro-select:157]: The `sorted` function doesn't change the original list, so these lines seem to have no effect.

Comment (timyuer) [reply bigtop-packages/src/common/bigtop-select/distro-select]: > The `sorted` function doesn't change the original list, so these lines seem to have no effect.

Thank you for your feedback. It has been modified.

**Comment by timyuer** [comment]
As https://github.com/apache/bigtop/pull/1203  submitted, this PR will be closed.

File: bigtop-packages/src/common/bigtop-select/conf-select
Comment (JiaLiangC) [inline bigtop-packages/src/common/bigtop-select/conf-select:1]: is there any reason for remove conf-select?

File: bigtop-packages/src/rpm/bigtop-select/SPECS/bigtop-select.spec [hunk 2]
```diff
@@ -32,16 +32,15 @@ BuildRoot:  %(mktemp -ud %{_tmppath}/%{name}-%{version}-%{release}-XXXXXX)
...
+Source2:    distro-select
 Requires:   bash
 
 # "which" command is needed for a lot of projects.
 # It is part of the package "util-linux" on suse and "which" everywhere else
 %if  %{?suse_version:1}0
 Requires:  util-linux
 %else
-Requires:       which
+Requires:       which, python3
```
Comment (JiaLiangC) [inline bigtop-packages/src/rpm/bigtop-select/SPECS/bigtop-select.spec:43]: At present, all Python dependencies have been removed from Bigtop, and it is now up to the user to manage them. This seems to be redundant here.

**Review by JiaLiangC — APPROVED** [review]

**Comment by JiaLiangC** [comment]
LGTM If there are no further review comments,  will be merged in 24 hours later.

**Pull request merged**